src/features/device-config/device-config.ts

```diff
@@ -49,7 +49,13 @@ export const generateConfig = async (
 			// Older ones have to use the old "user api keys"
 			return await createUserApiKey(req, (await userPromise).id);
 		}
-		return await createProvisioningApiKey(req, app.id);
+		return await createProvisioningApiKey(
+			req,
+			app.id,
+			req.body.provisioningKeyName
+				? { name: req.body.provisioningKeyName }
```

Comment:
You should probably also add the ability to name the provisioning api key whilst going via the `/api-key/application/:appId/provisioning` and `/api-key/v1` routes, cc @thgreasi

Reply:
Makes sense. `/api-key/v1` already has this available via body `{ name: "key-name" }`, but the `/api-key/application/:appId/provisioning` needed that in place.

Added it here.